subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -527,8 +527,52 @@ uint8_t ll_adv_aux_set_count_get(void)
 
 uint8_t ll_adv_aux_set_remove(uint8_t handle)
 {
-	/* TODO: reset/release primary channel and Aux channel PDUs */
-	return 0;
+	struct ll_adv_set *adv;
+	struct lll_adv *lll;
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+	struct lll_adv_aux *lll_aux;
+#endif/* !CONFIG_BT_CTLR_ADV_EXT */
+
+		/* Get the advertising set instance */
+	adv = ull_adv_is_created_get(handle);
+	if (!adv) {
+		return BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER;
+	}
+
+	if (adv->is_enabled) {
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
+#if defined(CONFIG_BT_CTLR_ADV_PERIODIC)
+	lll = &adv->lll;
+	if (lll->sync) {
+		struct lll_adv_sync *lll_sync;
+		struct ll_adv_sync_set *sync;
+
+		lll_sync = lll->sync;
+		sync = (void *)HDR_LLL2EVT(lll_sync);
+
+		if (sync->is_enabled) {
+			return BT_HCI_ERR_CMD_DISALLOWED;
+		}
+	}
+#endif /* CONFIG_BT_CTLR_ADV_PERIODIC */
+
+	lll = &adv->lll;
```

Comment:
Redundant assignment if line 547 is compiled in. Move to line above 546 and remove from here.

Reply:
Fixed